CMakeLists.txt

```diff
@@ -18,7 +18,19 @@ project(syslog-ng C)
 
 option(ENABLE_CPP "Enable C++" ON)
 
+if (ENABLE_CPP)
+  enable_language(CXX)
+  set(CMAKE_CXX_STANDARD 14 CACHE STRING "C++ standard")
+  set(CMAKE_CXX_STANDARD_REQUIRED OFF CACHE BOOL "C++ standard is a requirement")
+  set(CMAKE_CXX_EXTENSIONS OFF CACHE BOOL "C++ compiler specific extensions")
+  set(SYSLOG_NG_ENABLE_CPP ${ENABLE_CPP})
+endif()
+
 if(APPLE)
+  option(ENABLE_DARWIN_OSL "Enable OSLog" ON)
```

Comment:
i suggest adding a new variable as well here e.g.

ENABLE_OBJC

or s.g. like that, and set it based on the status of ENABLE_DARWIN_OSL (and later also based on any optional module-enabled settings that depend on objc)

option(ENABLE_OBJC "Enable OSLog" ENABLE_DARWIN_OSL)